bundles/admin/admin-layereditor/view/AdminLayerForm/VisualizationTabPane/TimeSeries.jsx

```diff
@@ -0,0 +1,173 @@
+                    {metadataOptions.map((option) => (
+                        <Option key={option.value} value={option.value}>
+                            {option.name}
+                        </Option>
+                    ))}
+                </Select>
+            </StyledFormField>
+        </Fragment>
+    );
+};
+
+MetadataLayerSelect.propTypes = {
+    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
+    controller: PropTypes.instanceOf(Controller).isRequired,
+    onChange: PropTypes.func.isRequired,
+};
+
+const MetadataAttribute = ({ disabled, controller }) => (
+    <Fragment>
+        <Message messageKey='timeSeries.metadataAttribute' />
+        <StyledFormField>
+            <TextInput
+                disabled={disabled}
+                type='text'
+                onChange={(evt) => controller.setTimeSeriesMetadataAttribute(evt.target.value)}
+            />
+        </StyledFormField>
+    </Fragment>
+);
+
+MetadataAttribute.propTypes = {
+    disabled: PropTypes.bool.isRequired,
+    controller: PropTypes.instanceOf(Controller).isRequired,
+};
+
+const DEFAULT_METADATA_TOGGLE_LEVEL = 8;
+
+const SliderContainer = styled('div')`
+    padding-left: 15px;
+    padding-right: 15px;
+    padding-top: 10px;
+    padding-bottom: 10px;
+
+    .ant-slider-mark-text {
+        font-size: 11px;
+    }
+`;
+
+const MetadataToggleLevel = ({ disabled, scales, controller }) => {
+    const minZoomLevel = -1;
+    const maxZoomLevel = scales.length - 1;
+    const marks = {
+        [minZoomLevel]: <Message messageKey='timeSeries.noToggle' />,
+        [maxZoomLevel]: String(maxZoomLevel)
+    };
+    scales.forEach((scale, index) => {
+        if (index % 5 === 0) {
+            marks[index] = String(index);
+        }
+    });
+    return (
+        <Fragment>
+            <Message messageKey='timeSeries.metadataToggleLevel' />
+            <SliderContainer>
+                <Slider
+                    disabled={disabled}
+                    step={1}
+                    min={-1}
+                    max={scales.length - 1}
+                    marks={marks}
+                    defaultValue={DEFAULT_METADATA_TOGGLE_LEVEL}
+                    onChange={(value) => controller.setTimeSeriesMetadataToggleLevel(value)}
+                />
+            </SliderContainer>
+        </Fragment>
+    );
+};
+
+MetadataToggleLevel.propTypes = {
+    disabled: PropTypes.bool.isRequired,
+    scales: PropTypes.arrayOf(PropTypes.number).isRequired,
+    controller: PropTypes.instanceOf(Controller).isRequired,
+};
+
+const TimeSeriesContainer = styled('div')`
+  border: 1px solid #d9d9d9;
+  padding: 10px;
+  margin-top: 5px;
+  margin-bottom: 10px;
+`
+
+export const TimeSeries = ({ scales, controller }) => {
+    const [timeSeriesUI, setTimeSeriesUI] = useState(TIME_SERIES_UI.PLAYER);
+    const [metadataLayer, setMetadataLayer] = useState('');
```

Comment:
You shouldn't need to use state here as these are passed through controller -> the component is re-rendered with new props because of that. Instead take the options.timeseries as additional prop-value and render based on that.